### PR TITLE
Set parenthood in shallow nodes

### PR DESCRIPTION
**Pull request by ad-daniel**
**Description**

Fixes: https://github.com/cyberbotics/webots/issues/5058

CadShape nodes are essentially an assembly of multiple geometries and appearances. The geometries are static meshes but what makes CadShapes a bit special is that the list of appearances are initialized from an assimp material rather than the usual pipeline. As such, the resulting PBRApperances/ImageTextures are not fully fledged nodes but rather shallow ones, they have no surface-level existence and can't be interacted with, nor are they saved to file. The recent change to the way relative urls are resolved means that they need to at least have a parent defined so that we can climb back to the CadShape node, or `WbUrl::computePath` won't be able to resolve it.

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
Did you already try to set the WbNode::gParent variable before instantiating the appearance and image texture node instead of passing the parent node as an argument?

https://github.com/cyberbotics/webots/blob/master/src/webots/vrml/WbNode.hpp#L79

File: src/webots/vrml/WbNode.cpp [hunk 1]
```diff
@@ -172,12 +172,12 @@ void WbNode::init() {
 // special constructor for shallow nodes, it's used by CadShape to instantiate PBRAppearances from an assimp material in
 // order to configure the WREN materials. Shallow nodes are invisible but persistent, and due to their incompleteness should not
 // be modified or interacted with in any other way other than through the creation and destruction of CadShape nodes
-WbNode::WbNode(const QString &modelName, const aiMaterial *material) {
+WbNode::WbNode(const WbNode *parentNode, const QString &modelName, const aiMaterial *material) {
```
Comment (stefaniapedrazzi) [inline src/webots/vrml/WbNode.cpp:175]: This is not new, but the `const aiMaterial *material` argument is not used and should be removed.

**Comment by ad-daniel** [comment]
Both aspects can indeed be simplified, thank you for the suggestion

**Review by stefaniapedrazzi — APPROVED** [review]
Thank you!

**Pull request merged**